src/xunit.netcore.extensions/Discoverers/ConditionalTestDiscoverer.cs

```diff
@@ -54,11 +55,30 @@ internal class ConditionalTestDiscoverer
                 if (symbols.Length == 2)
                 {
                     conditionMemberName = symbols[1];
-                    ITypeInfo type = testMethod.TestClass.Class.Assembly.GetTypes(false).Where(t => t.Name.Contains(symbols[0])).FirstOrDefault();
+
+                    // First, try to find the method in the assembly of the test method.
+                    ITypeInfo type = GetTypeFromSymbol(testMethod.TestClass.Class.Assembly, symbols[0]);
                     if (type != null)
                     {
                         declaringType = type.ToRuntimeType();
                     }
+                    else
+                    {
+                        try
+                        {
+                            var assemblyName = new AssemblyName("CoreFx.Private.TestUtilities");
```

Comment:
I don't like us hardcoding the name of this assembly here. We need a more general mechanism for finding these methods. Perhaps we can look at all assemblies referenced or we add another options that allows folks to put the assembly name as part of the string name.